test/functional/wallet_txn_doublespend.py

```diff
@@ -139,5 +140,99 @@ def run_test(self):
+        # reason (either block or mempool)
+
+        # Get a "blind" address from node0
+        connect_nodes(self.nodes[0], 1)
+        node0_address_blind = self.nodes[0].getnewaddress()
+        txid_origin = self.nodes[1].sendtoaddress(self.nodes[1].getnewaddress(), Decimal("10"))
+        self.nodes[1].generate(1)
+        self.sync_blocks([self.nodes[0], self.nodes[1]])
+
+        # Build parent conflicted tx from node1 to node1
+        nA = next(tx_out["vout"] for tx_out in self.nodes[1].gettransaction(txid_origin)["details"] if tx_out["amount"] == Decimal("10"))
+        inputs_parent_conflicted_tx = []
+        inputs_parent_conflicted_tx.append({"txid": txid_origin, "vout": nA, "sequence": BIP125_SEQUENCE_NUMBER})
+        outputs_parent_conflicted_tx = {}
+        outputs_parent_conflicted_tx[self.nodes[1].getnewaddress()] = Decimal("9.99998")
+
+        # Broadcast parent conflicted tx
+        parent_conflicted_tx = self.nodes[1].signrawtransactionwithwallet(self.nodes[1].createrawtransaction(inputs_parent_conflicted_tx, outputs_parent_conflicted_tx))
+        parent_conflicted_txid = self.nodes[1].sendrawtransaction(parent_conflicted_tx["hex"])
+
+        # Build children conflicted tx
+        inputs_children_conflicted_tx = []
+        inputs_children_conflicted_tx.append({"txid": parent_conflicted_txid, "vout": 0})
+        outputs_children_conflicted_tx = {}
+        outputs_children_conflicted_tx[node0_address_blind] = Decimal("9.99996")
+
+        #Broadcast children conflicted tx
+        node1s_children_conflicted_tx = self.nodes[1].signrawtransactionwithwallet(self.nodes[1].createrawtransaction(inputs_children_conflicted_tx, outputs_children_conflicted_tx))
+        children_conflicted_txid = self.nodes[1].sendrawtransaction(node1s_children_conflicted_tx["hex"])
+        self.sync_mempools([self.nodes[0], self.nodes[1]])
+
+        # Verify node0 sees conflicted paying back
+        node0s_children_conflicted_tx = self.nodes[0].gettransaction(children_conflicted_txid)
+
+        assert_equal(node0s_children_conflicted_tx["confirmations"], 0)
+        assert_equal(node0s_children_conflicted_tx["details"][0]["category"], "receive")
+
+        # Build conflicting tx, double-spending parent conflicted
+        inputs_conflicting_tx = []
+        inputs_conflicting_tx.append({"txid": txid_origin, "vout": nA})
+        outputs_conflicting_tx = {}
+        outputs_conflicting_tx[self.nodes[1].getnewaddress()] = Decimal("9.99980")
+
+        # Broadcast conflicting tx to node1 mempool and generate a block
+        conflicting_tx= self.nodes[1].signrawtransactionwithwallet(self.nodes[1].createrawtransaction(inputs_conflicting_tx, outputs_conflicting_tx))
+        conflicting_txid = self.nodes[1].sendrawtransaction(conflicting_tx["hex"])
+        self.nodes[1].generate(1)
+        self.sync_blocks([self.nodes[0], self.nodes[1]])
+
+        # Verify node0 sees children as conflicted
```

Comment:
Some comments might be better as log statements, to aid debugability. See https://github.com/bitcoin/bitcoin/tree/master/test/functional#general-test-writing-advice
```suggestion
        self.log.info('Verify node0 sees children as conflicted')
```

Reply:
Can I turn everything comment as a log ? That would my personal flavor but that's hard to judge if it suits everyone. In practice, this advice maybe a bit loosely, I expect anyone to tweak tests as wish for debugging, that's easy and you may not want to observe same thing for everyone.

Reply:
I mean mostly for the comments that start a new "section" or "paragraph" or subtest/test case. In the failing logs you could then easily identify when the new test section begins.

Reply:
> I mean mostly for the comments that start a new "section" or "paragraph" or subtest/test case. In the failing logs you could then easily identify when the new test section begins.

Yes! and logging like this also makes it clear where the slower tests are to possibly speed them up.